### PR TITLE
Remove deleted filesets when doing a scan

### PR DESCRIPTION
**Pull request by servos**
Remove filesets that have deleted files when doing a scan.

### CODE REVIEW

**Comment by servos** [comment]
@iwanders Won't let me add you as a reviewer :(

**Comment by mikepurvis** [comment]
Looks like it was a new @iwanderscp account which got created and added to the github org. We can add @iwanders instead, but IT should be made aware so they can keep track of it.

File: marv/_aggregate.py [hunk 1]
```diff
@@ -189,8 +189,12 @@ def __getitem__(self, key):
         if key != 'meta' and not self.meta[key].has_output:
             return None
         node_json = os.path.join(self.path, '{}.json'.format(key))
-        with open(node_json) as f:
-            return json_load_agg(f)
+        try:
+            with open(node_json) as f:
+                return json_load_agg(f)
+        except IOError:
+            log.warn('File not found: %s', node_json)
```
Comment (iwanders) [inline marv/_aggregate.py]: Printing the error itself is useful, `IOError` is also raised on permission denied for example.

File: marv/_site.py [hunk 2]
```diff
@@ -196,6 +196,17 @@ def scan(self):
         ctx = ScanContext(self.aggroot, self.scanroot, self.settype, self.settypename,
                           self.render_listing, self.render_detail)
         ctx.scan()
+
+        for info in ctx.files_removed:
+            relpath = info.path[len(self.scanroot)+1:]
+            file = File.query.filter(File.relpath == relpath).first()
```
Comment (iwanders) [inline marv/_site.py]: Please don't use [file](https://docs.python.org/2/library/functions.html#file) as a variable name.

**Review by iwanders — APPROVED** [review]
Good idea to discard deleted filesets.

**Pull request merged**